processor/transformprocessor/internal/metrics/func_convert_summary_quantile_val_to_gauge.go

```diff
@@ -0,0 +1,75 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package metrics // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/transformprocessor/internal/metrics"
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottldatapoint"
+)
+
+type convertSummaryQuantileValToGaugeArguments struct {
+	AttributeKey ottl.Optional[string]
+}
+
+func newConvertSummaryQuantileValToGaugeFactory() ottl.Factory[ottldatapoint.TransformContext] {
```

Comment:
I may be missing something, but I think we should use the metric context for this function.

Reply:
If I recall correctly, this is essentially a holdover from before OTTL had any way to work on metrics directly. At a minimum, we should have this function work on metrics, but we should also implement a feature gate to move the other two.

Reply:
good then, will adapt the context here and file a separate PR for the feature gate

Reply:
should be adapted now, will post the PR with the context change with feature gate in the next days